src/rtc_memory_esp32.cc

```diff
@@ -128,6 +134,11 @@ static void reset_rtc(const char* reason) {
   memset(&rtc, 0, sizeof(rtc));
   // We only clear RTC on boot, so this must be exactly 1.
   rtc.boot_count = 1;
+
+#ifndef CONFIG_IDF_TARGET_ESP32
```

Comment:
So we want this on the S3 and C3 too?

Reply:
Add this because we should initialize hardware system timer before calling trying to get time, you can see here: https://github.com/toitware/esp-idf/blob/67fa2950f6bed9cc8e2e89a8ffac1ed77f087214/components/esp_timer/Kconfig#L54.

Reply:
Added a comment.

Reply:
@donghengqaz: is the call to esp_timer_impl_early_init documented somewhere?